modules/nw-ovn-kubernetes-rollback.adoc

```diff
@@ -0,0 +1,80 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/rollback-to-openshift-sdn.adoc
+
+[id="nw-ovn-kubernetes-rollback_{context}"]
+= Rolling back the network provider to OpenShift SDN
+
+As a cluster administrator, you can rollback the Pod network provider for your cluster to the OpenShift SDN Pod network provider.
+During the rollback, you must reboot every node in your cluster.
+
+[IMPORTANT]
+====
+Only rollback to OpenShift SDN if the migration to the OVN-Kubernetes Pod network provider is unsuccessful.
+====
+
+.Prerequisites
+
+* Install the OpenShift Command-line Interface (CLI), commonly known as `oc`.
+* Access to the cluster as a user with the `cluster-admin` role.
+* A cluster installed on bare metal infrastructure configured with the OVN-Kubernetes Pod network provider.
+
+. The OpenShift SDN network provider namespace must be deleted prior to the rollback. To remove the namespace, enter the following command:
++
+----
+$ oc delete namespace openshift-sdn
+----
+
+. To enable the migration, set an annotation on the Cluster Network Operator configuration object by entering the following command:
++
+----
+$ oc annotate Network.operator.openshift.io cluster \
+  'networkoperator.openshift.io/network-migration'=""
+----
+
+. To change the network provider, enter the following command:
++
+----
+$ oc patch Network.config.openshift.io cluster \
+  --type='merge' --patch '{ "spec": { "networkType": "OpenShiftSDN" } }'
+----
+
+. Optional: Use the backup of the cluster network configuration to restore any customizations to the network configuration that you might have made, such as the maximum transmission unit (MTU) value.
+
+. To confirm the migration disabled the OVN-Kubernetes network provider, enter the following command. It might take several moments for all the OVN-Kubernetes Pods to stop.
++
+----
+$ oc get pod -n openshift-ovn-kubernetes --watch
+----
+
+. To complete the rollback, reboot each node in your cluster. For example, the following bash script reboots each node in your cluster. The script assumes that you can connect to each host by using `ssh` and that you have configured `sudo` to not prompt for a password.
++
+[source,bash]
+----
+#!/bin/bash
+
+for ip in $(oc get nodes  -o jsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}')
+do
+   echo "reboot node $ip"
+   ssh -o StrictHostKeyChecking=no core@$ip sudo shutdown -r -t 3
+done
+----
+
+. After the nodes in your cluster have rebooted, enter the following command to confirm that the Pod network provider is OpenShift SDN:
++
+----
+$ oc describe clusteroperators/network
+----
+
+. To confirm that the OpenShift SDN Pods are in the `READY` state, enter the following command:
++
+----
+$ oc get pod -n openshift-sdn --watch
+----
+
+. To remove the migration annotation from the Cluster Network Operator configuration object, enter the following command:
```

Comment:
Users also need to delete the namespace openshift-ovn-kubernetes manually after the rollback.